components/modal/Modal.spec.ts

```diff
@@ -0,0 +1,178 @@
+import { render, fireEvent } from '@testing-library/vue'
+import { vi } from 'vitest'
+import Modal from './Modal.vue'
+import Button from '../button/Button.vue'
+
+it('should render properly without any props', () => {
+  const screen = render({
+    components: { Modal },
+    template  : `
+      <Modal>Modal</Modal>
+    `,
+  })
+
+  const modal = screen.queryByTestId('modal')
+  const text  = screen.queryByText('Modal')
+
+  expect(modal).toBeInTheDocument()
+  expect(modal).toHaveClass('modal')
+  expect(text).toBeInTheDocument()
+})
+
+it('should have no close button if props "dismissable" set to false', () => {
+  const screen = render({
+    components: { Modal },
+    template  : `
+      <Modal :dismissable="false">
+        Modal
+      </Modal>
+    `,
+  })
+
+  const dismiss = screen.queryByTestId('modal-dismiss')
+
+  expect(dismiss).not.toBeInTheDocument()
+})
+
+it('should emit event "close" if close button clicked', async () => {
+  const spy    = vi.fn()
+  const screen = render({
+    components: { Modal },
+    template  : `
+      <Modal
+        title="Modal Title"
+        @close="onDismissed">
+        Modal Text
+      </Modal>
+    `,
+    methods: { onDismissed: spy },
+  })
+
+  const modal = screen.queryByTestId('modal')
+  const close = screen.queryByTestId('modal-dismiss')
+
+  expect(modal).toBeInTheDocument()
+  expect(close).toBeInTheDocument()
+  expect(spy).not.toBeCalled()
+
+  await fireEvent.click(close)
+
+  expect(spy).toBeCalled()
+})
+
+it('Modal will close if `Escape` button was pressed', async () => {
+  const screen = render({
+    components: { Modal },
+    template  : `
+      <Modal
+        title="Modal Title">
+        Modal Text
+      </Modal>
+    `,
+  })
+
+  const modal = screen.queryByTestId('modal')
+  const text  = screen.queryByText('Modal Text')
+
+  expect(modal).toBeInTheDocument()
+  expect(text).toBeInTheDocument()
+
+  await fireEvent.keyDown(window, { key: 'Escape' })
+
+  expect(modal).toHaveAttribute('style', 'display: none;')
```

Comment:
@avriq bisa pake `expect(modal).not.toBeVisible()` buat ngecek element hidden atau gak